lib/mongo/topology_description.ex

```diff
@@ -11,8 +11,8 @@ defmodule Mongo.TopologyDescription do
   alias Mongo.ReadPreference
 
   # see https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#topologydescription
-  @type type :: :unknown | :single | :replica_set_no_primary |
-                :replica_set_with_primary | :sharded
+  @type type :: :"unknown()" | :"single()" | :"replica_set_no_primary()" |
+                :"replica_set_with_primary()" | :"sharded()"
```

Comment:
You are right.
I don't know why the compiler was checking this typespec on the first argument of `TopologyDescription.select_servers/3`

And, it was calling a function on 
[test/mongo/topology_description_test.rb](https://github.com/ankhers/mongodb/blob/master/test/mongo/topology_description_test.exs) 
`TopologyDescription.select_servers(single(), :read, opts)`

Shouldn't be set as private `@typep type`?

I'm sorry, I'm too newbie on this, and thanks for the patience!